simtools/applications/tune_psf.py

```diff
@@ -313,22 +318,24 @@ def run_pars(pars, plot=True):
     # Running the tuning for all parameters in all_parameters
     # and storing the best parameters in best_pars
     min_rmsd = 100
+    best_pars = None
     for pars in all_parameters:
         _, rmsd = run_pars(pars, plot=args_dict["plot_all"])
         if rmsd < min_rmsd:
             min_rmsd = rmsd
             best_pars = pars
-
-    # Rerunning and plotting the best pars
-    run_pars(best_pars, plot=True)
-
-    plt.close()
-    pdf_pages.close()
-
-    # Printing the results
-    print("Best parameters:")
-    for par, value in best_pars.items():
-        print(f"{par} = {value}")
+    if best_pars is not None:
```

Comment:
Same thing there, as run_pars apparently cannot handle best_pars == None, the error handling should happen there.

Reply:
Changed